cloudinit/sources/DataSourceAzure.py

```diff
@@ -773,14 +784,20 @@ def exc_cb(msg, exception):
         return return_val
 
     @azure_ds_telemetry_reporter
-    def _report_ready(self, lease):
-        """Tells the fabric provisioning has completed """
+    def _report_ready(self, lease: dict) -> bool:
+        """Tells the fabric provisioning has completed.
+
+        @param lease: dhcp lease to use for sending the ready signal.
+        @return: The success status of sending the ready signal.
+        """
         try:
             get_metadata_from_fabric(None, lease['unknown-245'])
+            return True
         except Exception as e:
```

Comment:
To address any potential concerns: This works, since `get_metadata_from_fabric` always raises an exception within the implemented child functions of `get_metadata_from_fabric`.

[code](https://github.com/canonical/cloud-init/blob/3b05b1a6c58dfc7533a16f795405bda0e53aa9d8/cloudinit/sources/helpers/azure.py#L914-L922)
[code](https://github.com/canonical/cloud-init/blob/72f1d8df96a9e14444f96363c9b08845bbfe265b/tests/unittests/test_datasource/test_azure_helper.py#L744-L751)
[code](https://github.com/canonical/cloud-init/blob/0d5ac8f035c78971f7857a67bf9ce637a6798c77/tests/unittests/test_datasource/test_azure_helper.py#L709-L727)